Framework/HistogramData/src/Histogram.cpp

```diff
@@ -140,6 +159,9 @@ void Histogram::setSharedX(const Kernel::cow_ptr<HistogramX> &x) & {
 
   Throws if the size does not match the current size. */
 void Histogram::setSharedY(const Kernel::cow_ptr<HistogramY> &y) & {
+  if (yMode() == YMode::Uninitialized)
+    throw std::logic_error(
+        "Histogram::setSharedY: YMode is not set and cannot be determined");
```

Comment:
I have actually considered putting the `YMode` flag into `HistogramY`. That would allows us to increase the number of points where we have safety checks. What do you think? This could be done as a follow up at a later point.